nova/scheduler/filters/shard_filter.py

```diff
@@ -114,11 +120,72 @@ def _get_shards(self, project_id):
 
         return self._PROJECT_SHARD_CACHE.get(project_id)
 
-    def host_passes(self, host_state, spec_obj):
+    def _get_k8s_cluster_instances(self, spec_obj):
+        """If the instance will be part of a K8S cluster, it returns
+        the list of all other instances that are already part of it,
+        if any.
+        """
+        # BigVMs are scheduled based on their own rules.
+        if nova_utils.is_big_vm(spec_obj.memory_mb, spec_obj.flavor):
+            return []
```

Comment:
hm ... if we ignore big VMs here, but later explicitly handle `hana_*` flavors ... I mean big VMs are always `hana_*` flavors

I think we should ignore `hana_*` flavors here already and handle them explicitly in a follow-up, as we also have a ticket open on handling them differently.

Reply:
I mean we do not go into the `_host_passes_k8s()` for hana_* flavors and just handle non-hana VMs here for now, as we will implement something afterwards for hanas. IIRC, we will implement "ignore the shard filter for hana flavors unless the project explicitly wants to be bound to a shard for hanas".